hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -0,0 +1,244 @@
+    except:
+        exit("Cannot open file")
+    
+    try:
+        pinName_dict = pinName_to_dict(pinName_content)
+    except:
+        exit("Cannot extract PinName enum from file")
+
+    report = []
+    errors = []
+
+    if args.generic or not (args.generic or args.arduino):
+        identity_errors = identity_assignment_check(pinName_dict)
+        if identity_errors:
+            report.append(['', 'Generic pin names', 'identity', 'FAILED', len(identity_errors)])
+        else:
+            report.append(['', 'Generic pin names', 'identity', 'PASSED', 0])
+        errors += identity_errors
+
+        nc_errors = nc_assignment_check(pinName_dict)
+        if nc_errors:
+            report.append(['', 'Generic pin names', 'NC', 'FAILED', len(nc_errors)])
+        else:
+            report.append(['', 'Generic pin names', 'NC', 'PASSED', 0])
+        errors += nc_errors
+
+        duplicate_errors = duplicate_assignment_check(pinName_dict)
+        if duplicate_errors:
+            report.append(['', 'Generic pin names', 'duplicate', 'FAILED', len(duplicate_errors)])
+        else:
+            report.append(['', 'Generic pin names', 'duplicate', 'PASSED', 0])
+        errors += duplicate_errors
+
+        legacy_errors = legacy_assignment_check(pinName_content)
+        if legacy_errors:
+            report.append(['', 'Generic pin names', 'legacy', 'FAILED', len(legacy_errors)])
+        else:
+            report.append(['', 'Generic pin names', 'legacy', 'PASSED', 0])
+        errors += legacy_errors
+
+    if args.arduino or not (args.generic or args.arduino):
+        arduino_existence_errors = arduino_existence_check(pinName_dict)
+        if arduino_existence_errors:
+            report.append(['', 'Arduino Uno', 'existence', 'FAILED', len(arduino_existence_errors)])
+        else:
+            report.append(['', 'Arduino Uno', 'existence', 'PASSED', 0])
+        errors += arduino_existence_errors
+
+        arduino_nc_errors = arduino_nc_assignment_check(pinName_dict)
+        if arduino_nc_errors:
+            report.append(['', 'Arduino Uno', 'NC', 'FAILED', len(arduino_nc_errors)])
+        else:
+            report.append(['', 'Arduino Uno', 'NC', 'PASSED', 0])
+        errors += arduino_nc_errors
+
+        arduino_duplicate_errors = arduino_duplicate_assignment_check(pinName_dict)
+        if arduino_duplicate_errors:
+            report.append(['', 'Arduino Uno', 'duplicate', 'FAILED', len(arduino_duplicate_errors)])
+        else:
+            report.append(['', 'Arduino Uno', 'duplicate', 'PASSED', 0])
+        errors += arduino_duplicate_errors
+    
+    if errors:
+        print(tabulate(errors, headers=['Pin', 'Value', 'Error'], tablefmt='fancy_grid'))
+    
+    print(tabulate(report, headers=['Platform name', 'Test suite', 'Test case', 'Result', 'Errors'], tablefmt='fancy_grid'))
+
+    if errors:
+        print("Errors found in PinNames file")
+        exit(1)
+    else:
+        print("PinNames file is valid")
```

Comment:
```suggestion
        print("%s file is valid" % args.path")
```

Reply:
logging error wouldn't be nice in this case rather than print with prefix error/warning, etc?